src/router/index.js

```diff
@@ -6,6 +6,19 @@ const { routeConfigInjector } = require('@middlewares/routeConfigInjector')
 const { rateLimiter } = require('@middlewares/rateLimiter')
 const bodyParser = require('body-parser')
 const { httpMethods } = require('@constants/httpMethods')
+// Function to handle JSON parsing errors
+const handleJsonParsingError = (req, res, next) => {
+    bodyParser.json({ limit: '50MB' })(req, res, err => {
+        if (err instanceof SyntaxError) {
```

Comment:
Check if(err && err.type === 'entity.parse.failed') is intercepting the same error or not. This is just for information since this is the error type mentioned in the documentation.